tls/s2n_server_hello.c

```diff
@@ -286,9 +286,6 @@ int s2n_server_hello_recv(struct s2n_connection *conn)
         POSIX_GUARD(s2n_prf_key_expansion(conn));
     }
 
-    /* Choose a default signature scheme */
-    POSIX_GUARD(s2n_choose_default_sig_scheme(conn, &conn->handshake_params.server_cert_sig_scheme, S2N_SERVER));
```

Comment:
This isn't needed because `s2n_signature_algorithm_recv()` now selects a pre-TLS 1.2 default in `s2n_server_key_recv()`, right? And `server_cert_sig_scheme` doesn't need to be set to a TLS 1.2 default?

Reply:
Correct. See https://github.com/aws/s2n-tls/pull/4259#discussion_r1372219424